internal/grpc/interceptors/eventsmiddleware/events.go

```diff
@@ -234,6 +234,8 @@ func publisherFromConfig(m map[string]interface{}) (events.Publisher, error) {
 			EnableTLS:            m["enable-tls"].(bool),
 			TLSInsecure:          m["tls-insecure"].(bool),
 			TLSRootCACertificate: tlsCert,
+			AuthUsername:         m["username"].(string),
+			AuthPassword:         m["password"].(string),
```

Comment:
this is a little bit fragile ... if the config struct `m` does not have a key set this will panic with
```
"interface conversion: interface {} is nil, not string"
```
same for all the other keys, here ... still kinda meh.

Reply:
Good point :+1: I moved the definition to the source struct and use `mapstructure.Decode` to unmarshal it. Please double check.